src/pcre2_auto_possess.c

```diff
@@ -1115,6 +1117,12 @@ for(;;)
       break;
 #endif
 
+      case OP_ECLASS:
+      if (PRIV(eclass)(chr,
+          (list_ptr == list ? code : base_end) - list_ptr[2] + LINK_SIZE,
+          (list_ptr == list ? code : base_end) - list_ptr[3], utf)) return FALSE;
```

Comment:
Interesting. There are five instances of `[^ ]\?` and ten instances of ` \?` in this particular file.

It seems to be formatted consistently as `list_ptr == list ? code : base_end` throughout the file, so I'll keep it for now, rather than do extra reformatting.